shared/stack/stack_service.go

```diff
@@ -77,6 +76,12 @@ func (s *StackService) WithWorkspaceRepo(workspaceRepo workspacerepo.WorkspaceRe
 	return s
 }
 
+func (s *StackService) WithContext(env, appName string) *StackService {
```

Comment:
This function is misleading, because you don't pass in a context object. Why is this moved from the constructor? Both are required arguments for StackService.

Reply:
One of the implementations has a WithHappyConfig(), which was removed and replaced with the new builder method. I can rename it to WithEnvContext. Passing these arguments into a constructor conflicts with the builder approach, which only see arguments get added over time.

Reply:
The builder approach still applies, but if the arguments are required, then we should keep them in the constructor. Otherwise you might build a bad object

Reply:
Here's a list of args passed through builder calls:

WithBackend
WithExecutor
WithWorkspaceRepo

None of them are present in the constructor args. Creating a constructor with 5 args makes a messy constructor.

Reply:
We can convert them to options.

Reply:
But I'd prefer to do that in a separate PR.